### PR TITLE
Add a JakartaEE and MP example

### PR DESCRIPTION
**Pull request by gkwan-ibm**
Add an example demonstrates LangChain4J in a Jakarta EE / MicroProfile application.

### CODE REVIEW

**Comment by gkwan-ibm** [comment]
hi @langchain4j 

I submitted this PR a couple weeks ago. Let me know whatever need to be changed. Thanks.

**Review by langchain4j — APPROVED** [review]
@gkwan-ibm Thanks a lot for this!

**Comment by gkwan-ibm** [comment]
hi @langchain4j

Thank you to give me all the comments. I updated the example. Let me know any more code need to be changed.

File: jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/chat/ChatAgent.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package dev.langchain4j.example.chat;
+
+public interface ChatAgent {
+
+    String chat(String userMessage);
```
[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/chat/ChatAgent.java]: Updated the code to use `@ChatMemoryId`.

File: jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("language")
+    @Operation(
+        summary = "Use the language model.",
+        description = "Provide a sequence of words to a large language model.",
+        operationId = "languageModelAsk" )
+    public String languageModelAsk(@QueryParam("question") String question) {
+
+        HuggingFaceLanguageModel model = HuggingFaceLanguageModel.builder()
```
[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java]: Updated the code to create `HuggingFaceLanguageModel` once.

File: jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            .accessToken(HUGGING_FACE_API_KEY)
+            .modelId(TII_UAE_FALCON_7B_INSTRUCT)
+            .timeout(ofSeconds(120))
+            .temperature(1.0)
+            .maxNewTokens(200)
+            .waitForModel(true)
+            .build();
+
+        SystemMessage agentMessage =
+            systemMessage("I am knowledgeable about Large Language Models. How can I help you?");
```
[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java]: Updated the code to use the suggested message.

File: jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        HuggingFaceChatModel model = HuggingFaceChatModel.builder()
+            .accessToken(HUGGING_FACE_API_KEY)
+            .modelId(TII_UAE_FALCON_7B_INSTRUCT)
+            .timeout(ofSeconds(120))
+            .temperature(1.0)
+            .maxNewTokens(200)
+            .waitForModel(true)
+            .build();
+
+        SystemMessage agentMessage =
```
[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java]: Thanks to let me know the differences.

File: jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        summary = "Use the embedding model.",
+        description = "Determine the similarity and relevance score of two sentences.",
+        operationId = "similarity" )
+    public Properties similarity(
+               @QueryParam("text1") String text1,
+               @QueryParam("text2") String text2) {
+
+        HuggingFaceEmbeddingModel model = HuggingFaceEmbeddingModel.builder()
+                .accessToken(HUGGING_FACE_API_KEY)
+                .modelId(SENTENCE_TRANSFORMERS_ALL_MINI_LM_L6_V2)
```
[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java:133]: Tried `AllMiniLmL6V2EmbeddingModel`. When I updated the code or microprofile-config.properties file, I got following exception:
```
java.lang.UnsatisfiedLinkError: Native Library /Users/gkwan/.djl.ai/tokenizers/0.15.0-0.26.0-osx-x86_64/libtokenizers.dylib already loaded in another classloader
[INFO] 	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:208)
[INFO] 	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:181)
[INFO] 	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:1772)
[INFO] 	at java.base/java.lang.System.load(System.java:704)
[INFO] 	at ai.djl.huggingface.tokenizers.jni.LibUtils.loadLibrary(LibUtils.java:76)
[INFO] 	at ai.djl.huggingface.tokenizers.jni.LibUtils.<clinit>(LibUtils.java:41)
[INFO] 	at ai.djl.huggingface.tokenizers.HuggingFaceTokenizer.newInstance(HuggingFaceTokenizer.java:173)
[INFO] 	at dev.langchain4j.model.embedding.OnnxBertBiEncoder.<init>(OnnxBertBiEncoder.java:39)
[INFO] 	at dev.langchain4j.model.embedding.AbstractInProcessEmbeddingModel.loadFromJar(AbstractInProcessEmbeddingModel.java:20)
[INFO] 	at dev.langchain4j.model.embedding.AllMiniLmL6V2EmbeddingModel.<clinit>(AllMiniLmL6V2EmbeddingModel.java:18)
[INFO] 	at dev.langchain4j.example.rest.ModelResource.similarity(ModelResource.java:131)
[INFO] 	at
```

[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java:133]: So, I didn't update the code to use `AllMiniLmL6V2EmbeddingModel`.

[the remark this answers is not in the document]
Comment (gkwan-ibm) [reply jakartaee-microprofile-example/src/main/java/dev/langchain4j/example/rest/ModelResource.java]: opened issue #66

**Comment by GraceJansen** [comment]
Hi @langchain4j, is there anything else needed to get this PR merged? We have a blog we'd love to publish on this but we were waiting until this was merged into the main fork. If there's anything else you need from us for this, please let us know. 
@gkwan-ibm

File: jakartaee-microprofile-example/pom.xml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            <groupId>org.eclipse.microprofile</groupId>
+            <artifactId>microprofile</artifactId>
+            <version>6.1</version>
+            <type>pom</type>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j</artifactId>
+            <version>0.28.0</version>
```
Comment (langchain4j) [inline jakartaee-microprofile-example/pom.xml:35]:
```suggestion
            <version>0.29.0</version>
```

File: jakartaee-microprofile-example/pom.xml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        </dependency>
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j-hugging-face</artifactId>
+            <version>0.28.0</version>
```
Comment (langchain4j) [inline jakartaee-microprofile-example/pom.xml:40]:
```suggestion
            <version>0.29.0</version>
```

**Review by langchain4j — DISMISSED** [review]
@gkwan-ibm thanks a lot! And pardon for the late review, I have too much on my plate right now...

**Pull request merged**